libr/io/p/io_uf2.c

```diff
@@ -0,0 +1,454 @@
+			// or debug info that does not fit on the device flash
+			continue;
+		}
+
+		if ((block.flags & FILE_CONTAINER) != 0) {
+			// It is also possible to use the UF2 format as a container for one or more
+			// regular files (akin to a TAR file, or ZIP archive without compression).
+			// This is useful when the embedded device being flashed sports a file system.
+			// The field fileSize holds the file size of the current file,
+			// and the field targetAddr holds the offset in current file.
+			// The file name is stored at &data[payloadSize] and terminated with a 0x00.
+			ut8 *file_name = (ut8 *)(block.data + block.payloadSize);
+			R_LOG_WARN ("uf2: Found FILE_CONTAINER flag @ block #%d, TODO"
+					"{ file_name: \"%s\", chunk: %d, total_size: %d, offset: %d }", block.blockNo,
+					file_name, block.payloadSize, block.fileSize, block.targetAddr);
+			continue;
+		}
+
+		if ((block.flags & FAMILY_ID) != 0) {
+			// R_LOG_DEBUG ("uf2: Found FAMILY_ID flag @ block #%d", block.blockNo);
+			bool new_family_id = family_id != block.fileSize;
+			if (new_family_id) {
+				if (family_id != 0) {
+					R_LOG_WARN ("uf2: FAMILY_ID 0x%08x changed", block.fileSize);
+				}
+				family_id =  block.fileSize;
+				process_family_id (core, family_id);
+			}
+		}
+
+		if ((block.flags & MD5_CHECKSUM) != 0) {
+			R_LOG_WARN ("uf2: Found MD5_CHECKSUM flag @ block #%d, TODO", block.blockNo);
+			// the last 24 bytes of data[] hold the following structure:
+			// Offset | Size | Value
+			// 0      | 4    | Start address of region
+			// 4      | 4    | Length of region in bytes
+			// 8      | 16   | MD5 checksum in binary format
+		}
+
+		if ((block.flags & EXTENSION_TAGS) != 0) {
+			// Extension tags can, but don't have to, be repeated in all blocks.
+			R_LOG_WARN ("uf2: Found EXTENSION_TAGS flag @ block #%d", block.blockNo);
+			ut16 offset = block.payloadSize;
+			while (offset < 476) {
+				ut32 tag_sz = *(ut32 *)(block.data + offset);
+				ut32 tag = (tag_sz >> 8) & 0xffffff;
+				ut8 size = (ut8)(tag_sz & 0xff);
+				switch (tag) {
+				case ETAG_FW_VERSION:
+				case ETAG_FW_CHECKSUM:
+				case ETAG_DESCRIPTION:
+				case ETAG_DEVICE_ID:
+				case ETAG_PAGE_SIZE:
+				default:
+					R_LOG_WARN ("uf2: Unknown EXTENSION TAG 0x%04x, TODO", tag);
+					// TODO dump size
+					break;
+				}
+				// Every tag starts at 4 byte boundary
+				offset += pad (size, 4);
+			}
+		}
+
+		if (r_buf_write_at (rbuf, block.targetAddr, block.data, block.payloadSize) != block.payloadSize) {
+			R_LOG_ERROR ("uf2: Sparse buffer problem, giving up");
+			return false;
+		}
+
+		if (has_debug) {
+			R_LOG_DEBUG ("uf2: Block #%02d (%d bytes @ 0x%08x)", block.blockNo, block.payloadSize, block.targetAddr);
+		}
+
+		comment = r_str_newf ("uf2 block #%02d (%d bytes @ 0x%08x)\n", block.blockNo, block.payloadSize, block.targetAddr);
+		r_meta_set_string (core->anal, R_META_TYPE_COMMENT, block.targetAddr, comment);
```

Comment:
use `io->coreb.cmdf (core, comment);`

Reply:
where comment must start with `"CC ...` aka contain the command that adds a comment